numba/dppl/tests/dppl/test_ufuncs.py

```diff
@@ -0,0 +1,1879 @@
+import functools
+import itertools
+import re
+import sys
+import warnings
+import threading
+import operator
+
+import numpy as np
+
+import unittest
+from numba import typeof, njit
+from numba.core import types, typing, utils
+from numba.core.compiler import compile_isolated, Flags, DEFAULT_FLAGS
+from numba.np.numpy_support import from_dtype
+from numba import jit, vectorize
+from numba.core.errors import LoweringError, TypingError
+from numba.tests.support import TestCase, CompilationCache, MemoryLeakMixin, tag
+from numba.core.typing.npydecl import supported_ufuncs, all_ufuncs
+from numba.np import numpy_support
+from numba import dppl
+
+is32bits = tuple.__itemsize__ == 4
+iswindows = sys.platform.startswith('win32')
+
+# NOTE: to test the implementation of Numpy ufuncs, we disable rewriting
+# of array expressions.
+
+enable_pyobj_flags = Flags()
+enable_pyobj_flags.set("enable_pyobject")
+enable_pyobj_flags.set("no_rewrites")
+
+no_pyobj_flags = Flags()
+no_pyobj_flags.set("no_rewrites")
```

Comment:
This flag might prevent rewrite of numpy calls to array expressions, which in turn will prevent us to generate parfor nodes which actually triggers the gpu code generation. Please make sure that is the case and we will have to think about removing this flag in that case.